packages/drivers/routerlicious-driver/src/wholeSummaryDocumentStorageService.ts

```diff
@@ -141,8 +148,24 @@ export class WholeSummaryDocumentStorageService implements IDocumentStorageServi
         return summaryHandle;
     }
 
-    public async downloadSummary(handle: ISummaryHandle): Promise<ISummaryTree> {
-        throw new Error("NOT IMPLEMENTED!");
+    public async downloadSummary(summaryHandle: ISummaryHandle): Promise<ISummaryTree> {
+        const wholeFlatSummary = await PerformanceEvent.timedExecAsync(
+            this.logger,
+            {
+                eventName: "getWholeFlatSummary",
+                treeId: summaryHandle.handle,
+            },
+            async (event) => {
+                const response = await this.manager.getSummary(summaryHandle.handle);
+                event.end({
+                    size: response.trees[0]?.entries.length,
+                });
+                return response;
+            },
+        );
+
+        const {blobs, snapshotTree} = this.convertWholeFlatSummaryToSnapshotTreeAndBlobs(wholeFlatSummary, false);
```

Comment:
As per my comment here https://github.com/microsoft/FluidFramework/pull/9882/files#r850772702, this should probably call the existing `convertWholeFlatSummaryToSnapshotTreeAndBlobs` and return this output. The caller should then add the ".app" tree if required. The reason is that I believe this API should return the summary in the same format as it was uploaded which is without the summary wrapped in ".app" tree.

The issue is that this feature is trying to re-use the `rehydrateDetachedFromSnapshot` API which expects the summary to have the ".app" tree since that is how its generated by `serialize`.

Reply:
For the sake of moving this PR through. I may separate conversion/download into its own PR, and tag you there @agarwal-navin

Reply:
Moved out downloadSummary and conversion code, so we can have focused conversation around these other items.